src/pool.h

```diff
@@ -31,8 +31,12 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 #ifndef POOL_H
 #define POOL_H
+#define MAX_RIG_ID 32
+#define MAX_RIGS 1024
```

Comment:
You're also currently defining MAX_RIG_ID twice (header and implementation) and these should be removed from here anyway if you use start/end pointers to the buffer.